### PR TITLE
Issue #346 improve installation instructions

### LINKED ISSUE #346: Improve installation instructions
Following many discussions and related to issue #215, we need to improve the installation instructions to reflect prerequisites and additional manual steps that aren't caused by bugs (meaning they are expected).

Submitting a PR on this in 3... 2... 1...

### PR DESCRIPTION
**Pull request by thatdocslady**
Fixes #346 

This PR restructures and tweaks the installation instructions to (hopefully) clarify where manual steps might be required, point better to additional information, and generally clean up the flow of information to help users get started more easily. 

Related to issue #215

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,13 +13,15 @@ provisioning VMs, and [OpenShift Origin](https://github.com/openshift/origin) fo
...
-    - [Installing via Homebrew](#installing-via-homebrew)
+    - [Installing with Homebrew](#installing-with-homebrew)
 - [Quickstart](#quickstart)
   - [Starting Minishift](#starting-minishift)
   - [Deploying a sample application](#deploying-a-sample-application)
   - [Reusing the Docker daemon](#reusing-the-docker-daemon)
 - [Documentation](#documentation)
 - [Known Issues](#known-issues)
+  - [Bugs](#bugs)
+  - [Unsupported features](#unsupported-features)
```
Comment (hferentschik) [inline README.md]: What's 'unsupported' features!?

File: README.md [hunk 2]
```diff
@@ -41,40 +43,43 @@ Minishift requires a hypervisor to run the virtual machine containing OpenShift.
...
+Make sure that the hypervisor is installed and enabled on your system before you install Minishift.
 
-Drivers can be selected via the `--vm-driver=xxx` flag of `minishift start` as mentioned in the [Starting Minishift](#starting-minishift) section below. See the [Troubleshooting](/docs/troubleshooting.md#kvm-drivers) guide, in case you encounter any issues.
+**Important:**
 
-**Note:**
-- For most hypervisors, VT-x/AMD-v virtualization must be enabled in the BIOS. For Hyper-V, however,
-it needs to be disabled.
-- We recommend that you use `Virtualbox >= 5.1.12` on Windows to avoid the issue
+- On some operating systems, KVM and xhyve require specific installation and configuration steps.
```
Comment (hferentschik) [inline README.md]: This reads to me, that KVM or xhyve would be an option on other than Linux and OS X respectively. But KVM is the Linux hypervisor and xhyve OS X. They don't "install" on other OS types.

File: README.md [hunk 3]
```diff
@@ -41,40 +43,43 @@ Minishift requires a hypervisor to run the virtual machine containing OpenShift.
...
+For more information, see [docker machine drivers installation](./docs/docker-machine-drivers.md).
+- For most hypervisors, you must enable VT-x/AMD-v virtualization in the BIOS. However, if you want
```
Comment (hferentschik) [inline README.md]: I think this is actually not true. So we should most likely delete this.

File: README.md [hunk 4]
```diff
@@ -41,40 +43,43 @@ Minishift requires a hypervisor to run the virtual machine containing OpenShift.
...
 
+If you encounter driver issues, see the [Troubleshooting](/docs/troubleshooting.md) guide.
+
 <a name="installing-minishift"></a>
 ### Installing Minishift
 
-Download the archive matching your host OS from the Minishift [releases page](https://github.com/minishift/minishift/releases) and unpack it. Copy the contained binary to your preferred
-location and optionally ensure it is added to your _PATH_.
+1. Download the archive for your operating system from the Minishift [releases page](https://github.com/minishift/minishift/releases) and unpack it.
+2. Copy the contents of the directory to your preferred location and add the path to your _PATH_ environment variable.
```
Comment (hferentschik) [inline README.md]: Are this not really 3 steps, where the last one is optional?

Comment (hferentschik) [inline README.md]: Why do you not use '1.' all the time. Markdown will do the numbering?

File: README.md [hunk 5]
```diff
@@ -41,40 +43,43 @@ Minishift requires a hypervisor to run the virtual machine containing OpenShift.
...
-- Automatic update of minishift binary and virtual machine ISO is currently disabled, due to issues
- [#204](https://github.com/minishift/minishift/issues/204),
- [#178](https://github.com/minishift/minishift/issues/178),
- [#112](https://github.com/minishift/minishift/issues/112) and
- [#192](https://github.com/minishift/minishift/issues/192). We will take a comprehensive look at
- these issues in an upcoming release and provide an improved solution for automatic updates.
 
-<a name="installing-via-homebrew"></a>
-#### Installing via Homebrew
+- On Windows operating system, due to issue [#236](https://github.com/minishift/minishift/issues/236), you need to execute the minishift binary
```
Comment (hferentschik) [inline README.md]: Just "On Windows" !?

File: README.md [hunk 6]
```diff
@@ -41,40 +43,43 @@ Minishift requires a hypervisor to run the virtual machine containing OpenShift.
...
+from the drive containing your %USERPROFILE% directory.
+- Automatic update of the Minishift binary and the virtual machine ISO is currently disabled, due to the following issues:
+  - [#204](https://github.com/minishift/minishift/issues/204)
```
Comment (hferentschik) [inline README.md]: I think we should just link to the main issue around this and not list all of them.

File: README.md [hunk 7]
```diff
@@ -106,7 +111,9 @@ The interaction with OpenShift is via the command line tool _oc_ which is copied
 <a name="starting-minishift"></a>
 ### Starting Minishift
 
-1. Assuming you have put _minishift_  on the _PATH_ as described in [Installing Minishift](#installing-minishift) you can start Minishift via:
+1. Run the `minishift start` command. The following example shows the result of
+the command when you start Minishift on a Linux operating system with the KVM
```
Comment (hferentschik) [inline README.md]: why do we use Linux as an example here? Why not windows or OS X? That are out bigger target groups.

File: README.md [hunk 8]
```diff
@@ -122,28 +129,32 @@ The interaction with OpenShift is via the command line tool _oc_ which is copied
...
 
-    Note that, the IP seen above is dynamic and can change. It can be retrieved with `minishift ip`. Also,
-    instead of 'kvm', you will see 'xhyve' on Mac OS and 'hyperv' on Windows.
-
-    **Note:** By default Minishift uses the driver most relevant to the host OS.
-To use a driver of choice for Minishift, use the `--vm-driver=xxx` flag with `minishift start`. For example, to use VirtualBox instead of KVM for Fedora, use `minishift start --vm-driver=virtualbox`.
+  **Note**:
 
-1. Add `oc` binary to the _PATH_:
+  - The IP is dynamically generated for each Minishift instance. To check the IP, run the `minishift ip` command.
```
Comment (hferentschik) [inline README.md]: "Minishift instance"!? Not sure whether this is the right terminology here.

File: README.md [hunk 9]
```diff
@@ -122,28 +129,32 @@ The interaction with OpenShift is via the command line tool _oc_ which is copied
...
 
         $ export PATH=$PATH:~/.minishift/cache/oc/v1.3.1
 
-1. Login to your OpenShift account and authenticate yourself:
+  **Note:** Depending on the operating system and the `oc` version, you might need
+  to use a different command to add the path to the _PATH_ environment variable.
+  To verify the `oc` version, check the contents of the `~/.minishift/cache/oc` directory.
 
-        $ oc login https://192.168.99.128:8443 -u developer -p developer
+3. You can access the OpenShift Web console from your Web browser by navigating to the IP address listed in the CLI with the same credentials that you use to log
```
Comment (hferentschik) [inline README.md]: That's not correct. You switched what's happening here. The original example was driving from the console logging in on the console/terminal using oc. You are now describing a completely different thing which is access to the OpenShift console. Also Minishift has 'minishift console' to open the OpenShift console in the default browser for you.

File: README.md [hunk 10]
```diff
@@ -122,28 +129,32 @@ The interaction with OpenShift is via the command line tool _oc_ which is copied
...
+into the command-line console.
 
 <a name="deploying-a-sample-application"></a>
 ### Deploying a sample application
 
-You can use Minishift to run a sample Node.js application on OpenShift as follows:
+You can use Minishift to run a sample Node.js application on OpenShift from the
+Web console or from the command-line. OpenShift provides various sample builder
+applications, quickstarts, and so on.
```
Comment (hferentschik) [inline README.md]: "and so on"!?

File: README.md [hunk 11]
```diff
@@ -201,6 +212,14 @@ The following documentation is available:
 <a name="known-issues"></a>
 ## Known Issues
 
+<a name="bugs"></a>
+### Bugs
+
+For a list of current bugs, see the [release notes](https://github.com/minishift/minishift/releases).
```
Comment (hferentschik) [inline README.md]: -1, the release notes don't list the current bugs. It is the issue tracker which has the current bugs. The release notes, refer to bugs/issues fixed.

File: README.md [hunk 12]
```diff
@@ -201,6 +212,14 @@ The following documentation is available:
...
+
+<a name="unsupported-features"></a>
+### Unsupported features
```
Comment (hferentschik) [inline README.md]: I think we have to say this different. "Unsupported features" to me means, there is something which one can use which is not supported by some sort of support contract. This does not make sense from a community perspective and is not really what this is about.

File: README.md (hunk 1 above)
Comment (Preeticp) [inline README.md]: My 2 cents, the Known Issues section need not be a part of the Readme at all, especially now, with 'Bugs' being made into a sub-heading which shows in the TOC. It's like a user, right at the onset is being told, here is a product and here are its bugs. I do not say we need to hide them but we do not need to prominently display them either.
We could have a seperate doc with known issues, like here: https://github.com/sorccu/cufon/wiki/Known-bugs-and-issues or https://github.com/dotnet/core/tree/master/cli. 
We could put a line with a link in the Community section and also in the Contributing guidelines saying, before you raise issues please have a look at this doc which lists the known issues, or such.

File: README.md [hunk 13]
```diff
@@ -122,28 +129,32 @@ The interaction with OpenShift is via the command line tool _oc_ which is copied
...
-1. Add `oc` binary to the _PATH_:
+  - The IP is dynamically generated for each Minishift instance. To check the IP, run the `minishift ip` command.
+  - By default Minishift uses the driver most relevant to the host OS. To use a different driver, set the `--vm-driver` flag in `minishift start`. For example, to use VirtualBox instead of KVM on Linux operating systems, run `minishift start --vm-driver=virtualbox`. For more information about `minishift start` options,
+  see the [minishift start command reference](/docs/minishift_start.md).
 
-     **Note:** How to modify the _PATH_ varies depending on host OS, version of the OC binary and
-     other variables. In case of doubt, you can check the content of the
-     `~/.minishift/cache/oc` directory.
+2. Add the `oc` binary to the _PATH_ environment variable. For example, on Linux
+operating systems, run the following command:
```
Comment (Preeticp) [inline README.md]: The first step says its a Linux example, so  'For example on Linux operating systems' again here, seems unnecessary, specially when it is followed by a note saying it could be different for other Operating systems.

File: README.md (hunk 1 above)
Comment (hferentschik) [inline README.md]: > My 2 cents, the Known Issues section need not be a part of the Readme at all, especially now, with 'Bugs' being made into a sub-heading which shows in the TOC.

+1, known issues is ok imo, but really what we try to express are "limitations". Regarding to "bugs" I feel exactly the way. No need to highlight this in this way. Also, in its current form the link does not even point to the issue tracker.

File: README.md (hunk 13 above)
Comment (hferentschik) [inline README.md]: +1

File: README.md (hunk 1 above)
Comment (thatdocslady) [inline README.md]: This section was already in the README, I just separated it out and gave it a title.

File: README.md (hunk 2 above)
Comment (thatdocslady) [inline README.md]: Oh sorry, I'll change it to distro. Didn't notice this.

File: README.md (hunk 3 above)
Comment (thatdocslady) [inline README.md]: I had to enable virtualization in my BIOS before I could run minishift, but honestly I don't remember which type it was. I just tabbed thru all the options until I saw "virtualization"

File: README.md (hunk 4 above)
Comment (thatdocslady) [inline README.md]: - I'll break step 2 to two steps
- It doesn't matter really but I can change it

File: README.md (hunk 5 above)
Comment (thatdocslady) [inline README.md]: The text already said this was only for Windows, I just restructured the note to state the OS first so that other OS users won't need to read most of the sentence to find out it's for Windows. If this issue is now relevant for other OSs let me know and I'll change it.

File: README.md (hunk 6 above)
Comment (thatdocslady) [inline README.md]: which is the main issue here?

File: README.md (hunk 8 above)
Comment (thatdocslady) [inline README.md]: you're right, I'll change to OpenShift cluster

File: README.md (hunk 7 above)
Comment (thatdocslady) [inline README.md]: I notice that you're asking some questions regarding the text accuracy, which I cannot answer, since I didn't touch so much the existing content but rather tried to make the flow of information more clear and do a general proofread. Some information might be outdated here but it's hard for me to determine this. If you want to request updated info feel free to write it in and I'll fix it, but for example in this case I didn't decide to use Linux as the example text and I don't know what the syntax for Windows/MacOS.

File: README.md (hunk 9 above)
Comment (thatdocslady) [inline README.md]: The original step to log into the console was deemed redundant yesterday when @Preeticp was testing things, so we said we will delete it. Since the output of minishift start specifically states that you can log in to the console by navigating to the URL, I thought I would write this here. If it's unrelated to the quickstart I can remove it.

File: README.md (hunk 10 above)
Comment (thatdocslady) [inline README.md]: Yes, there are multiple types of sample apps. I can rephrase it to list them as examples and get rid of the trailing text.

File: README.md (hunk 11 above)
Comment (thatdocslady) [inline README.md]: Aha ok, so it only lists fixed bugs. We need to find a way to mention currently open bugs (which translate to known issues), but that's tracked in a different issue. For now I'll remove this.

File: README.md (hunk 12 above)
Comment (thatdocslady) [inline README.md]: Like I mentioned, the text was already there so I just gave it a title. If you have a suggestion on how to frame this, based on your knowledge as to whether we plan to support these or not, please suggest it.

File: README.md (hunk 1 above)
Comment (thatdocslady) [inline README.md]: In the beginning I myself suggested this but was pushed back because (I can't remember who said this) these limitations or known issues were supposedly important for the users to know before they get started. In general, once we have a publication flow, we should move these to a more appropriate place depending on what they represent.

File: README.md (hunk 13 above)
Comment (thatdocslady) [inline README.md]: will adjust

File: README.md (hunk 5 above)
Comment (hferentschik) [inline README.md]: issue #236 should do

File: README.md (hunk 6 above)
Comment (hferentschik) [inline README.md]: issue #236 should do

File: README.md (hunk 5 above)
Comment (hferentschik) [inline README.md]: this comment seems out of context. I might have placed it wrong, but with the old version gone, I am not quite sure what I wanted to comment

File: README.md [hunk 14]
```diff
@@ -198,8 +205,10 @@ The following documentation is available:
 * [Installing docker-machine drivers](./docs/docker-machine-drivers.md)
 * [Developing Minishift](./docs/developing.md)
 
-<a name="known-issues"></a>
-## Known Issues
+You can also view release notes in the [releases page](https://github.com/minishift/minishift/releases).
```
Comment (hferentschik) [inline README.md]: Why not add this to the enumeration?

File: README.md [hunk 15]
```diff
@@ -122,28 +129,28 @@ The interaction with OpenShift is via the command line tool _oc_ which is copied
...
-     other variables. In case of doubt, you can check the content of the
-     `~/.minishift/cache/oc` directory.
+1. Add the `oc` binary to the _PATH_ environment variable.
 
         $ export PATH=$PATH:~/.minishift/cache/oc/v1.3.1
 
-1. Login to your OpenShift account and authenticate yourself:
-
-        $ oc login https://192.168.99.128:8443 -u developer -p developer
+  **Note:** Depending on the operating system and the `oc` version, you might need
```
Comment (hferentschik) [inline README.md]: Now you removed the oc login completely!?

Comment (Preeticp) [inline README.md]: @hferentschik when you do `minishift start`, in the output, it states that you are logged in as developer, with password developer. I tested with the help of instructions from @praveenkumar, using `oc status` after I did `Minishift start`, without loging in again and verified that we did not need to login again. So we all agreed that this step could be done away with. Let us know if you have different thoughts about this.

Comment (hferentschik) [inline README.md]: > when you do minishift start, in the output, it states that you are logged in as developer, with password developer.

where does it do that? I have not seen that

> I tested with the help of instructions from @praveenkumar, using oc status after I did Minishift start, without loging in again and verified that we did not need to login again.

That's because you where logged in before and your _oc_ session had not expired yet. If you start from scratch also deleting/edting your ~/.kube/config  or letting the access token expire, you will need to login. Login is very much required. There is an issue for improving this and letting minishift log you in - https://github.com/minishift/minishift/issues/349, but right now there is no such thing.

> So we all agreed that this step could be done away with.

I must have missed this discussion

> Let us know if you have different thoughts about this.

See above.

Comment (praveenkumar) [inline README.md]: I apologize , @hferentschik  is right if `.kube` file present then it take credentials from there otherwise we need to login so yes it's required step.

Comment (Preeticp) [inline README.md]: when you do minishift start, in the output, it states that you are logged in as developer, with password developer.
where does it do that? I have not seen that

Well, @hferentschik, when I do `minishift start` this is the output I get:
$ minishift start
Starting local OpenShift instance using 'kvm' hypervisor...
Provisioning OpenShift via '/home/preeti/.minishift/cache/oc/v1.3.1/oc [cluster up --use-existing-config --host-config-dir /var/lib/minishift/openshift.local.config --host-data-dir /var/lib/minishift/hostdata]'
-- Checking OpenShift client ... OK
-- Checking Docker client ... OK
-- Checking Docker version ... OK
-- Checking for existing OpenShift container ... 
   Deleted existing OpenShift container
-- Checking for openshift/origin:v1.3.1 image ... OK
-- Checking Docker daemon configuration ... OK
-- Checking for available ports ... OK
-- Checking type of volume mount ... 
   Using Docker shared volumes for OpenShift volumes
-- Creating host directories ... OK
-- Finding server IP ... 
   Using 192.168.42.232 as the server IP
-- Starting OpenShift container ... 
   Creating initial OpenShift configuration
   Starting OpenShift using container 'origin'
   Waiting for API server to start listening
   OpenShift server started
-- Installing registry ... OK
-- Installing router ... OK
-- Importing image streams ... OK
-- Importing templates ... OK
-- Login to server ... OK
-- Creating initial project "myproject" ... OK
-- Server Information ... 
   OpenShift server started.
   The server is accessible via web console at:
       https://192.168.42.232:8443

   **You are logged in as:
       User:     developer
       Password: developer

   To login as administrator:
       oc login -u system:admin**

I tested with the help of instructions from @praveenkumar, using oc status after I did Minishift start, without logging in again and verified that we did not need to login again.

That's because you where logged in before and your oc session had not expired yet. If you start from scratch also deleting/edting your ~/.kube/config or letting the access token expire, you will need to login. Login is very much required. There is an issue for improving this and letting minishift log you in - #349, but right now there is no such thing.

In that case the last 5 lines of the output for `minishift start` is misleading, and if this (output) login is about something different, then it is not clear what it is for? At the first line of the output itself we speak about starting an OC instance, so it is natural for user to assume that this for OC.

@praveenkumar np, I feel, we are just trying to reduce ambiguities and experimenting our way into making the Minishift experience better. We can always add the step back if it's a required one, which leaves us with the confusing output message.

File: README.md [hunk 16]
```diff
@@ -99,17 +103,20 @@ You can now install the latest beta version of minishift.
 <a name="quickstart"></a>
 ## Quickstart
 
-This section contains a brief demo of Minishift and the provisioned OpenShift instance.
+This section contains a brief demo of Minishift and the provisioned OpenShift cluster.
 For details on the usage of Minishift refer to the [Using Minishift](/docs/using.md) guide.
 The interaction with OpenShift is via the command line tool _oc_ which is copied to your host.
 
+The following steps describe how to get started with Minishift on a Linux operating system
```
Comment (LalatenduMohanty) [inline README.md]: We should write GNU/Linux instead of just Linux.

File: README.md (hunk 15 above)
Comment (hferentschik) [inline README.md]: > **You are logged in as:
> User: developer
> Password: developer

Funny, which version of minishift are you using. I don't even see this. But that's besides the point really.

> In that case the last 5 lines of the output for minishift start is misleading, 

No, it might be that you are logged in, but my point is that this is a side effect of being logged in before. If you for example would do a _oc logout_ you will also see that login is required.

> At the first line of the output itself we speak about starting an OC instance, so it is natural for user to assume that this for OC.

Now, that's not true. Nowhere in the output it says that "OC" would be started. We are starting/provisioning OpenShift.

> We can always add the step back if it's a required one, 

+1

> which leaves us with the confusing output message.

Not at all, all is consistent. Also, in this tutorial we assume a clean install. Maybe for completeness sake we can add _oc logout_ somewhere at the end of the creating app example. But to be honest, no one does that. You want _oc_ to stay logged in, otherwise you would have to provide credentials for each and every single command you are running.

Comment (hferentschik) [inline README.md]: > I apologize , @hferentschik is right if .kube file present then it take credentials from there otherwise we need to login so yes it's required step.

no worries.

**Comment by LalatenduMohanty** [comment]
@hferentschik @praveenkumar @Preeticp 

Looks like the automatic login i.e. ~/.kube/config stuff happens as part of `minishift start` and we do not need to login explictly. I have not tracked the code responsible for this yet. 

```
$ export PATH=$PATH:~/.minishift/cache/oc/v1.4.1

#Logging out and removing the ~/.kube/config

$ oc logout
Logged "developer" out on "https://192.168.64.7:8443"

$ minishift delete
Deleting the Minishift VM...
Minishift VM deleted.           

$ rm -rf ~/.kube/
```
```
$ minishift start
Starting local OpenShift cluster using 'xhyve' hypervisor...
Provisioning OpenShift via '/Users/lmohanty/.minishift/cache/oc/v1.4.1/oc [cluster up --use-existing-config --host-config-dir /var/lib/minishift/openshift.local.config --host-data-dir /var/lib/minishift/hostdata]'
-- Checking OpenShift client ... OK
-- Checking Docker client ... OK
-- Checking Docker version ... OK
-- Checking for existing OpenShift container ... OK
-- Checking for openshift/origin:v1.4.1 image ... 
   Pulling image openshift/origin:v1.4.1
   Pulled 0/3 layers, 3% complete
   Pulled 1/3 layers, 71% complete
   Pulled 2/3 layers, 82% complete
   Pulled 3/3 layers, 100% complete
   Extracting
   Image pull complete
-- Checking Docker daemon configuration ... OK
-- Checking for available ports ... OK
-- Checking type of volume mount ... 
   Using Docker shared volumes for OpenShift volumes
-- Creating host directories ... OK
-- Finding server IP ... 
   Using 192.168.64.7 as the server IP
-- Starting OpenShift container ... 
   Creating initial OpenShift configuration
   Starting OpenShift using container 'origin'
   Waiting for API server to start listening
   OpenShift server started
-- Adding default OAuthClient redirect URIs ... OK
-- Installing registry ... OK
-- Installing router ... OK
-- Importing image streams ... OK
-- Importing templates ... OK
-- Login to server ... OK
-- Creating initial project "myproject" ... OK
-- Removing temporary directory ... OK
-- Server Information ... 
   OpenShift server started.
   The server is accessible via web console at:
       https://192.168.64.7:8443

   You are logged in as:
       User:     developer
       Password: developer

   To login as administrator:
       oc login -u system:admin

$ oc whoami
developer

$ oc status
In project My Project (myproject) on server https://192.168.64.7:8443

You have no services, deployment configs, or build configs.
Run 'oc new-app' to create an application.
```

File: README.md (hunk 15 above)
Comment (Preeticp) [inline README.md]: which version of minishift are you using. I don't even see this.
>@hferentschik, I am on minishift version: 1.0.0-beta.2, so probably that's why.
No, it might be that you are logged in, but my point is that this is a side effect of being logged in before. 
>Ok, that might explain why I see this, thank you.

**Comment by LalatenduMohanty** [comment]
As @praveenkumar pointed out looks like `oc cluster up` covers this. My experiment with `oc cluster up --create-machine`  confirms the same thing . Refer: https://github.com/openshift/origin/blob/master/pkg/bootstrap/docker/up.go#L879-L882

**Comment by Preeticp** [comment]
Ah Lala, just saw your comment, I had version 1.0.0-beta.2.

**Comment by thatdocslady** [comment]
@LalatenduMohanty @hferentschik @praveenkumar so now if I understand correctly we can remove the step to log into oc because it's handled as a part of minishift start (with oc cluster up under the hood)? Just want to clarify so that we're all in agreement here.

File: README.md (hunk 6 above)
Comment (thatdocslady) [inline README.md]: Your comment originally referred to automatic updates (which list multiple issues), while issue #236 is related to Windows drivers. Please clarify.

**Comment by hferentschik** [comment]
There seems to be a difference in behaviour between the first startup and successive restarts where it does not occur. The first startup seems to log you in indeed, not successive one though.

**Comment by thatdocslady** [comment]
@hferentschik ok so since the installation/quickstart doc aims to get people started the first time, maybe subsequent logins are out of scope. We can put in a note that says something like "autologin only happens the first time you start minishift, and in subsequent sessions you need to also log in"

**Comment by hferentschik** [comment]
> We can put in a note that says something like "autologin only happens the first time you start minishift, and in subsequent sessions you need to also log in"

We can, but I am reluctant, because moving forward we will do things from the Minishift side and once we add authentication features, the auto login might not even work anyways. I rather bnot expose this auto login at all atm.

**Comment by thatdocslady** [comment]
@hferentschik  ok so how do you recommend that we handle this? should we put the instruction back into the procedure even though it's redundant? what happens when users log in twice when they don't need to?

**Comment by hferentschik** [comment]
right, I'd leave it and logging in again does not matter at all.

**Comment by hferentschik** [comment]
>  ok so how do you recommend that we handle this?

let's leave this as is. There are a couple of other things though I'd like to suggest - https://github.com/thatdocslady/minishift/pull/1/files

File: README.md [hunk 17]
```diff
@@ -192,14 +199,15 @@ docker ps
 
 The following documentation is available:
 
+* [Releases Notes](https://github.com/minishift/minishift/releases).
 * [Using Minishift](./docs/using.md)
```
Comment (Preeticp) [inline README.md]: Releases Notes>Release Notes

Comment (hferentschik) [inline README.md]: +1, I added this to my pull request as well

**Comment by thatdocslady** [comment]
@hferentschik saw your edits, looks great thanks! If you can approve this pull request and merge it, I think Lala already did this, I guess it'll incorporate your commit as well?

**Comment by hferentschik** [comment]
> saw your edits, looks great thanks! 

cool

> If you can approve this pull request and merge it

You need to update your pull request first. If you are ok with what I suggest, you would merge my pull request into your branch and then push your branch again to your fork, so that the pull request gets updated.

> I guess it'll incorporate your commit as well?

Nope

**Comment by thatdocslady** [comment]
@hferentschik see that's what I'm asking, how can I update my PR with your commit? I remember you said at some point that we can both push to the same PR, so maybe we can do it like that?

**Comment by hferentschik** [comment]
> see that's what I'm asking, how can I update my PR with your commit? 

There are many ways, but the standard one would be as described in my comment https://github.com/minishift/minishift/pull/347#issuecomment-276911652.

That said, I went ahead and merged. If you are interested we can go through the workflow I described another day. I think it would be easier to explain this over bluejeans for example.

> I remember you said at some point that we can both push to the same PR

No, that we cannot do.

**Comment by hferentschik** [comment]
Merged.
